subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -0,0 +1,153 @@
+                       "Deferred-Delete count while cd'ed into container")
+
+        # cd out of the container filesystem. devmapper will notice and
+        # will clean up, but it may take a while. Allow up to 30 seconds.
+        os.chdir('/')
+        defer_count_back = lambda: self._defer_count() == defer_count_initial
+        self.failif(not utils.wait_for(defer_count_back, timeout=30),
+                    "Timed out waiting for Deferred-Delete count (%d) to"
+                    " come back down to %d after cd'ing out of container"
+                    % (self._defer_count(), defer_count_initial))
+
+    def _start_idle_container(self):
+        """
+        Start a container. We only need it briefly, until we (the test,
+        running in host space) can cd into its root filesystem. Container
+        will spin idly until a semaphore file is removed.
+        """
+        self.stuff['dc'] = dc = DockerContainers(self)
+        c_name = dc.get_unique_name()
+
+        fin = DockerImage.full_name_from_defaults(self.config)
+        subargs = ['--rm', '--name=' + c_name, fin,
+                   'bash -c "echo READY;touch /DELETE-ME;'
+                   ' while [ -e /DELETE-ME ]; do sleep 0.1; done"']
+        dkrcmd = AsyncDockerCmd(self, 'run', subargs)
+        dkrcmd.execute()
+        dkrcmd.wait_for_ready(c_name)
+        self.stuff['container_name'] = c_name
+
+    def _stop_idle_container(self):
+        """
+        Stop our idle container, by removing its semaphore file.
+        Wait until container is truly gone, as reported by docker ps -a
+        """
+        os.unlink('DELETE-ME')
+
+        dc = self.stuff['dc']
+        c_name = self.stuff['container_name']
+        c_gone = lambda: dc.list_containers_with_name(c_name) == []
+        self.failif(not utils.wait_for(c_gone, timeout=5),
+                    "Timed out waiting for container to exit")
+
+    @property
+    def _container_mountpoint(self):
+        """
+        Given a container name, return its host-accessible root filesystem
+        """
+        subargs = ['--format', '{{.GraphDriver.Data.DeviceName}}',
+                   self.stuff['container_name']]
+        inspect_cmd = mustpass(DockerCmd(self, 'inspect', subargs).execute())
+        devmap_entry = inspect_cmd.stdout.strip()
+
+        base_path = utils.run('findmnt --noheadings --output TARGET'
+                              ' --source /dev/mapper/%s' % devmap_entry)
+        mountpoint = os.path.join(base_path.stdout.strip(), 'rootfs')
+        self.logdebug('host mount point = %s', mountpoint)
+        return mountpoint
+
+    @staticmethod
+    def _defer_count():
+        """
+        Returns the integer count of deferred deletions, as obtained
+        from docker info. We assume DockerInfo() is never cached.
+        """
+        storage_fields = DockerInfo().sub_fields('Storage Driver')
+        return int(storage_fields['Deferred Deleted Device Count'])
```

Comment:
Okay, looks good'nuf.  I do think there should be a minimal ``cleanup()`` that tries some sort of half-arse job of recovering.  So, maybe if ``cd /``, ``rm -f DELETE-ME`` (ignore errors), ``self.stuff['dc'].clean_all()``?  Then it doesn't assume ``garbage_check`` is even enabled or running.